### ISSUE TITLE
Cleanup map compilation state

### ISSUE THREAD

**Issue by Whiteknight** [Whiteknight/CastIron#11]
`MapCompileOperation` is a mess, and it delegates to `ObjectCreatePreferences` which isn't used appropriately. `MapBuilder`/`IMapCompilerSettings` also only exposes options for a single data type, when we potentially want to configure constructor/factory settings for all data types used throughout the graph.

The `IMapCompilerSettings` interface needs to be redesigned, possibly with changes to the `IDataResults` interface to support a more flexible configuration system.

Basically, we want to build a config tree where Types[]->subtypes[]->options. Then we can include this whole thing in `MapCompileOperation` and cache/retrieve constructor/factory information for each type as it's encountered.

**Comment by Whiteknight** [comment]
This is all redone as TypeSettings->SpecificTypeSettings and is in much better condition than before. I'm closing this.
